
syntax = "proto3";
package imgpatch;


service ImgPatch {
  rpc ShowImgOnServer(Empty) returns (Empty) {}
  rpc SetPosition(Position) returns (Empty) {}
  rpc GetPixels(Empty) returns (PatchPixels) {}
  rpc SetPixels(PatchPixels) returns (Empty) {}
}


message Empty {}

message Position {
  int32 x_pos = 1;
  int32 y_pos = 2;
}

message PatchPixels {
  bytes pixels = 1;
}

